server/test/document.spec.js

```diff
@@ -0,0 +1,339 @@
+import supertest from 'supertest';
+import should from 'should';
+import app from '../../server';
+import testData from './helpers/specHelper';
+
+const server = supertest.agent(app);
+let token, adminToken, user4Token, userId, documentId1, documentId2;
+
+describe('Document Api', () => {
+  before((done) => {
+    server.post('/users')
+      .send(testData.regularUser3)
+      .end((err, res) => {
+        token = res.body.token;
+        userId = res.body.userId;
+      });
+    server.post('/users')
+      .send(testData.adminUser3)
+      .end((err, res) => {
+        should(res.body).have.property('token');
+        adminToken = res.body.token;
+      });
+    server.post('/users')
+      .send(testData.regularUser4)
+      .end((err, res) => {
+        should(res.body).have.property('token');
+        user4Token = res.body.token;
+        done();
+      });
+  });
+
+  describe('Create', () => {
+    it('should create new document', (done) => {
+      server.post('/documents')
+      .send(testData.document)
+      .set({ 'x-access-token': token })
+        .end((err, res) => {
+          res.body.should.have.property('document');
+          should(res.body.document.access).equal('public');
+          documentId1 = res.body.document.id;
+          done();
+        });
+    });
+
+    it('should not create document with null field(s)', (done) => {
+      const doc = {
+        title: 'Doc 1',
+      };
+      server.post('/documents')
+      .send(doc)
+      .set({ 'x-access-token': token })
+        .end((err, res) => {
+          should(res.body).have.a.property('error');
+          done();
+        });
+    });
+
+    it('should send an error message for a user not logged in', (done) => {
+      server.post('/documents')
+      .send(testData.document2)
+        .end((err, res) => {
+          res.status.should.equal(401);
+          done();
+        });
+    });
+
+    it('should be able to make document', (done) => {
+      server.post('/documents')
+      .send(testData.document2)
+      .set({ 'x-access-token': token })
+        .end((err, res) => {
+          res.body.should.have.property('document');
+          should(res.body.document.access).equal('private');
+          documentId2 = res.body.document.id;
+          done();
+        });
+    });
+  });
+
+  describe('Find', () => {
```

Comment:
The description here does adequately describe the test suites in this block. `Find` can be replaced with something like `Fetch Documents` as `Find` seems really vague.